### DISCUSSION TITLE
How to build and run the site locally?

### DISCUSSION THREAD

**Discussion by SScanry** [EliteMasterEric/Teyvat.moe#290] [Q&A] [answered]
Excuse me, I want to know how the code is executed. Because I have loaded the folder and tried to run it in visual studio code and it shows a window saying hello jerkin . I would like to know how to run it.

Thank you

**Comment by EliteMasterEric** [comment, marked as answer]
To run the site locally:

1. Download the site, obviously.
2. Download [NodeJS](https://nodejs.org/en/download/), a JavaScript runtime.
3. Navigate to the project directory, or open it in VSCode.
4. Open a terminal (in VSCode this is done with CTRL-SHIFT-` (the tilde key below the ESC)
5. Run `npm install`. This should install all the project dependencies into the `./node_modules` folder.
6. Run `npm run start:dev`. This should launch the site in development mode, accessible at `https://localhost:3000`.

If you want to build the site for production, run `npm run build` and the artifacts will be in the `./build` folder.

**Reply by vsatyamesc** [reply]
Hello, thank you for making this very helpful interactive map. Can you plz tell how can I run this as github website? I mean which branch should I choose and else

**Reply by vsatyamesc** [reply]
Hi I tried installing this through the above procedures but I'm getting a lot of error with the missing **package.json** I tried `npm init ` react is showing missing package name

**Reply by EliteMasterEric** [reply]
Sorry for the delayed response. Here is an updated procedure.

1. Download the site. Note that to do this properly you need to use a Git client to download the submodules like so:

`git clone --recursive https://github.com/Teyvat-Moe/Teyvat.moe`

2. Download NodeJS and preferably VSCode.
3. Open a terminal and call `yarn`. If you have the submodules installed as stated above, this should install all the required modules to `./node_modules`.
4. Running `yarn start:dev` will start the app in a local development mode and you can access it at `https://localhost:3000`.

The current state of the app is such that it does not support a full static page export (as is required to use with GitHub Pages). This is because the app uses NextJS's image optimization and this feature does not support static page export. I am using Netlify to host now (thus the switch to the new domain).

If you are experiencing issues with missing packages you probably did not setup the Git submodules. [Here is an article that can teach you about them](https://github.blog/2016-02-01-working-with-submodules/).

Also note that the Google sync feature will likely not work since it only functions on verified domains.
